### DISCUSSION TITLE
Geometrical segmentation (vertical & horizontal ) on point clouds

### DISCUSSION THREAD

**Discussion by HripsimeS** [pyvista/pyvista#4473] [Q&A] [answered]
### Describe the feature you would like to be added.

@eino @puripant @nschloe @leouieda @jourdain 

Hello. I would appreciate if you inform me if there is a way with PyVista to do a segmentation of horizontal and vertical point clouds. To get horizontal point clouds one color and other color for vertical point clouds.

I look forward to hearing from you and thanks in advance!!!

### Pseudocode or Screenshots

```python
Insert code here
```
Or drag your screenshot here!

**Comment by banesullivan** [comment]
Hi @HripsimeS, I have converted this topic to a Discussion since it's more Q&A than development oriented. 

Would you please take a moment to look at PyVista's documentation and examples to become familiar with the data model and common filtering and plotting routines? This will help you and us get started in solving this problem so that you can share code as a starting point to visualize the data you have: https://docs.pyvista.org/version/stable/getting-started/index.html

Then please share example data files and screenshots with annotations describing what you would like to accomplish.

Otherwise, this is a complex task and I don't think we'll be able to accomplish what's in your mind without a starting point and some data.

**Comment by HripsimeS** [comment]
@banesullivan thanks for your quick reply! In attached link you can find point cloud data for tree (green-yellow) and ground (red).
 https://drive.google.com/file/d/1AdB10T9M-DOF7P_bRMck_PMb241fDeD8/view?usp=drive_link

In addition I attached screenshots of visualization of my point cloud. I would like to segment separately the ground and the tree.
Unfortunately I don't have annotations for that, so I was thinking maybe possible to segment without labels to separate ground (horizontal to be red color) from the tree (vertical to be green color). Do you have any suggestions? :) 

![image](https://github.com/pyvista/pyvista/assets/42246765/f932fa7b-e12b-4bb1-a1f5-bcc75e795c8c)
![image](https://github.com/pyvista/pyvista/assets/42246765/b22bf4a0-e840-40f4-b391-81404f8a3071)

**Reply by banesullivan** [reply, marked as answer]
Thanks for sharing the data! Looking at this now, I think your best bet is to use a library like scikit-learn to perform clustering/segmentation then use PyVista for visualization and post-processing.

This looks like a great blog post on using scikit-learn to do a similar task: https://medium.com/@chimso1994/point-cloud-segmentation-in-python-2fdbf5ea0617

Sorry to deflect the question to another toolkit, but I think scikit learn is much better suited to this task and will ultimately make the task easier.

Hope this helps!

**Reply by HripsimeS** [reply]
@banesullivan thank you very much, I will try 🥇 :)

**Reply by banesullivan** [reply]
Let us know if you have visualization questions or want to filter/post-process the point cloud!
